routers/web/repo/issue_pin.go

```diff
@@ -33,22 +36,25 @@ func IssuePinOrUnpin(ctx *context.Context) {
 
 // IssueUnpin unpins a Issue
 func IssueUnpin(ctx *context.Context) {
-	issue, err := issues_model.GetIssueByIndex(ctx.Repo.Repository.ID, ctx.ParamsInt64(":id"))
+	issue, err := issues_model.GetIssueByID(ctx, ctx.ParamsInt64(":id"))
 	if err != nil {
-		ctx.Status(http.StatusNoContent)
+		ctx.Status(http.StatusInternalServerError)
+		log.Error(err.Error())
 		return
 	}
 
 	// If we don't do this, it will crash when trying to add the pin event to the comment history
 	err = issue.LoadRepo(ctx)
 	if err != nil {
 		ctx.Status(http.StatusInternalServerError)
+		log.Error(err.Error())
 		return
 	}
 
 	err = issue.Unpin(ctx, ctx.Doer)
 	if err != nil {
 		ctx.Status(http.StatusInternalServerError)
+		log.Error(err.Error())
```

Comment:
return?

Reply:
ps: What do you think about https://github.com/go-gitea/gitea/issues/24120 Handle function call errors more gracefully in routes  ? 

There are so many missing returns in many PRs. It wastes a lot of time for contributors.